plugins/push/frontend/public/templates/push-notification-drawer.html

```diff
@@ -0,0 +1,855 @@
+<cly-drawer
+    class="cly-vue-push-notification-drawer"
+    @submit="onSubmit"
+    @close="onClose"
+    @open="onOpen"
+    :title="title"
+    :saveButtonLabel="saveButtonLabel"
+    v-bind="controls">
+    <template v-slot:sidecars="drawerScope">
+        <cly-content id="mainSidecar" role="sidecar" :always-active="true" style="display: flex;justify-content: center;align-items: center; height:100%">
+            <mobile-message-preview
+                v-if="drawerScope.currentStepId === 'step3' ||  drawerScope.currentStepId === 'step4'"
+                :title="pushNotificationUnderEdit.message[activeLocalization].title" 
+                :content="pushNotificationUnderEdit.message[activeLocalization].content" 
+                :platforms="pushNotificationUnderEdit.platforms" 
+                :media="pushNotificationUnderEdit.message.mediaUrl" 
+                :isHTML="true"
+                :buttons="pushNotificationUnderEdit.message[activeLocalization].buttons.map(function(item){return item.label})">
+            </mobile-message-preview>
+        </cly-content>
+    </template>
+    <template v-slot:default="drawerScope">
+        <add-user-property-popover 
+            :isOpen="isAddUserPropertyPopoverOpen"
+            :width="350"
+            ref="addUserPropertyPopover"
+            :id="selectedUserPropertyId"
+            :container="selectedUserPropertyContainer"
+            :userProperty="pushNotificationUnderEdit.message[activeLocalization].properties[selectedUserPropertyContainer][selectedUserPropertyId]"
+            :options="userPropertiesOptions"
+            :position="addUserPropertyPopoverPosition"
+            @select="onSelectUserProperty"
+            @input="onInputFallbackUserProperty"
+            @check="onCheckUppercaseUserProperty"
+            @remove="onRemoveUserProperty"
+            @close="closeAddUserPropertyPopover">
+        </add-user-property-popover>
+        <div class="bu-column bu-is-10 bu-is-offset-1">
+            <cly-form-step id="step1" :name="i18n('push-notification.drawer-step-one')" @click="function(){console.log('onclick')}">
+                <div class="cly-vue-drawer-step__section">
+                    <div class="bu-py-1 bu-my-1 cly-vue-push-notification-drawer__input-label">Message Name</div>
+                    <div class="cly-vue-push-notification-drawer__input-description">We can add some kind of description for message name</div>
+                    <validation-provider v-slot="validation" rules="required" name="messageName">
```

Comment:
Just to be clear I wanted to use <cly-form-field> component which in turn would have reduced a lot of WET code when validating the data. Unfortunately, it has another section underneath and it messed up my layout. I did not think adding style on top of it was the right approach (push drawer layout differs by quite some margin from the rest of components layout)

Reply:
it is fine I guess, push is an exception to many things